mantidproject/mantid
# 24 context menu1 d hist ws

**Description of work.**

This PR added new plot menu (four types) in the context menu of 1D HistoWorkspace.





**To test:**
Compile MantidWorkBench using this branch, and add the following line to the interactive Ipython Console
```python
test = CreateMDHistoWorkspace(SignalInput='1,2,3,4,2,1',
                              ErrorInput='1,1,1,1,1,1',
                              Dimensionality=3,
                              Extents='-1,1,-1,1,0.5,6.5',
                              NumberOfBins='1,1,6',
                              Names='x,y,|Q|',
                              Units='mm,km,AA^-1')
```
Right click on `test` in the side panel `Workspaces` and you should see a new menu item plot, which would allow you to perform
* plot
* overplot
* plot with errors
* overplot with errors
![Screenshot from 2020-10-09 12-02-24](https://user-images.githubusercontent.com/5064852/95606110-e06a5780-0a27-11eb-99ee-043e55eec7bc.png)



This PR is corresponding to the this task
https://code.ornl.gov/sns-hfir-scse/spectroscopy/spectroscopy/-/issues/29

and its parent user story
https://code.ornl.gov/sns-hfir-scse/spectroscopy/spectroscopy/-/issues/24








---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
